node_tree.py

```diff
@@ -709,8 +766,16 @@ def sv_copy(self, original):
         
     def free(self):
```

Comment:
if a node is removed through the act of an undo, `free` is not called, this is why i added the lame solution to the undo handler. incase you were wondering :)

Reply:
The Blender undo system should pop a message saying "just undo if is the only choice". :)
If the `free ` function is not called then if there where any socket cache about that node it wont be deleted until the file is closed (as the no-users node-trees :) ) , not a big deal, just a secondary effect...

Reply:
it's a corner case. doesn't bother me, i consider it a bug that `free` is not called by blender, but i got nowhere when asking devs about it.

Reply:
i recall it was needed to remove stale drawing if the last node added was a VD